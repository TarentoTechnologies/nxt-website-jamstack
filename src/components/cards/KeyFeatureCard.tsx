import {
  bgCadiumOrange,
  bgPeacockGreen,
  bgSpectrumViolet,
  bgScarletRed,
  bgCeruleanBlue,
  bgItalianBlue,
} from "../../styles/style-guide/ColorGuide.module.css";
import { cardStyles, imgStyles } from "./KeyFeatureCard.module.css";

interface KeyFeatureCardProps {
  title: string;
  logo: any;
  bgColor: any;
  clickHandler?: (event: any) => void;
}

export const KeyFeatureCard = ({
  title,
  logo,
  bgColor,
  clickHandler,
}: KeyFeatureCardProps) => {
  const getBgColor = () => {
    switch (bgColor) {
      case "green":
        return bgPeacockGreen;
      case "red":
        return bgScarletRed;
      case "orange":
        return bgCadiumOrange;
      case "blue":
        return bgCeruleanBlue;
      case "italianBlue":
        return bgItalianBlue;
      case "violet":
        return bgSpectrumViolet;
      default:
        return "";
    }
  };
  const bgColour = getBgColor();
  return (
    <div
      className={`container-fluid col-sm-12 col-md-4 ${cardStyles} ${bgColour}`}
      onClick={clickHandler}
    >
      <div className={`${imgStyles}`}>
        <img src={logo} alt={title} title={title} className={`img-fluid`} />
      </div>
      <p>{title}</p>
    </div>
  );
};
