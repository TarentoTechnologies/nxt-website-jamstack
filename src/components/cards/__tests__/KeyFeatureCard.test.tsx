import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { KeyFeatureCard } from "../KeyFeatureCard";

const imgData = {
  localFile: {
    childImageSharp: {
      gatsbyImageData: {
        layout: "constrained",
        placeholder: {
          fallback:
            "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsTAAALEwEAmpwYAAACcklEQVR42rWV7UuTURjG958EUYRYWzr3PrfS5nTZdOhAMrFXIuhLIASVGKvcdHu26dame3FOl9tKxGKZ0tSWNZe1GQUSEfRB+ti/sKtzzhBEtrXKHrg5h4fn/O7rvq9zzsMTc0qjxK78KXUoIbUr8yTwh5GnaymDsnjSEdUPuVsFKfdXsEKQtYxBWDxCz/8TbBeUsnilPhDZZBBzchYirjCn71hwhbHYuqJAiV0BzcNmNLib2Njo1rJoGTsD3bgeWu9paD06yOz1lQHrbFKkvq1hYMEE92sPuBUHzC+H4V+fQCgTxmjKhex2jiRrYUp3g0uW7Fkbg95nYHOFU42TLg0sBOp9M46LM1dYoiaicm/pJUuOZmMwhrpQMyyC3KGCYKgOvrcB0Mf5yoXwxhR0pAW0mt8qpIBr8etMgcxRj+ME2jN9Hs8+J+BLB3B/yYzeyCWmmppVUiHNJOEUOGapxfynpzAEO3HQdIQo0SP9fZ1ALxClQmbQ4tYSmr2t5RXSbOrRRoTfTWP24xymNiKIZeOwJK1o9bVDaJOwfiqdJzCRmcQpsgMKphRRSLMIrRK2MPllGd3hXnRNdsO0+ACPPkRx4O4h1seqQT7razATgsZTxmX6stYqRpu/A7FcHHxSdvWgAGfDPfCng7gx14ehpA321RHcStzBwtaLyoGPN2eZEbSX54gZtLzL0au4nehH//MB9M3fhG3FDtVIQ2lTKJA2mJ6A3PYm4rkniLyfwerXFOvl4XtVzJCj5hrwyVhtFjADKzop1D1DwIj2QCfaAh3MqJ3tRINuJYVDXflZpn2hBu3E3rLKxX+4vvb7gt3vX8AvdXpaCCj9EUYAAAAASUVORK5CYII=",
        },
        images: {
          fallback: {
            src: "/static/285b5b87d3c327c3ce8d379e9c86f131/a8b52/tech_1_52cd05fc87.png",
            srcSet:
              "/static/285b5b87d3c327c3ce8d379e9c86f131/53973/tech_1_52cd05fc87.png 30w,\n/static/285b5b87d3c327c3ce8d379e9c86f131/7ab40/tech_1_52cd05fc87.png 60w,\n/static/285b5b87d3c327c3ce8d379e9c86f131/a8b52/tech_1_52cd05fc87.png 120w",
            sizes: "(min-width: 120px) 120px, 100vw",
          },
          sources: [],
        },
        width: 120,
        height: 120,
      },
    },
  },
  alternativeText: "Built for scale",
  caption: "Built for scale",
  id: "bfd0e978-24b6-57dd-a4e1-1c51d229a602",
};

describe("Test if the KeyFeatureCard component is working as expected", () => {
  it("<KeyFeatureCard /> matches snapshot", () => {
    const component = render(
      <KeyFeatureCard
        title={"Built for scale"}
        logo={imgData}
        bgColor={"green"}
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<KeyFeatureCard /> renders without exploding", () => {
    const component = render(
      <KeyFeatureCard
        title={"Built for scale"}
        logo={imgData}
        bgColor={"green"}
      />
    );
    expect(component.getByTestId("key-feature-card")).toBeInTheDocument();
  });
});
