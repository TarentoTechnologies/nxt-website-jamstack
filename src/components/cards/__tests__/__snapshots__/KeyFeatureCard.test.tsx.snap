// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the KeyFeatureCard component is working as expected <KeyFeatureCard /> matches snapshot 1`] = `
<div>
  <div
    class="container-fluid col-sm-12 col-md-6 col-lg-4 cardStyles bgPeacockGreen"
    data-testid="key-feature-card"
  >
    <div
      class="imgStyles"
    >
      <div
        class="gatsby-image-wrapper gatsby-image-wrapper-constrained img-fluid iconSize"
        data-gatsby-image-wrapper=""
        style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
      >
        <div
          style="max-width: 120px; display: block;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='120'%20width='120'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
            style="max-width: 100%; display: block; position: static;"
          />
        </div>
        <img
          alt=""
          aria-hidden="true"
          data-placeholder-image=""
          decoding="async"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsTAAALEwEAmpwYAAACcklEQVR42rWV7UuTURjG958EUYRYWzr3PrfS5nTZdOhAMrFXIuhLIASVGKvcdHu26dame3FOl9tKxGKZ0tSWNZe1GQUSEfRB+ti/sKtzzhBEtrXKHrg5h4fn/O7rvq9zzsMTc0qjxK78KXUoIbUr8yTwh5GnaymDsnjSEdUPuVsFKfdXsEKQtYxBWDxCz/8TbBeUsnilPhDZZBBzchYirjCn71hwhbHYuqJAiV0BzcNmNLib2Njo1rJoGTsD3bgeWu9paD06yOz1lQHrbFKkvq1hYMEE92sPuBUHzC+H4V+fQCgTxmjKhex2jiRrYUp3g0uW7Fkbg95nYHOFU42TLg0sBOp9M46LM1dYoiaicm/pJUuOZmMwhrpQMyyC3KGCYKgOvrcB0Mf5yoXwxhR0pAW0mt8qpIBr8etMgcxRj+ME2jN9Hs8+J+BLB3B/yYzeyCWmmppVUiHNJOEUOGapxfynpzAEO3HQdIQo0SP9fZ1ALxClQmbQ4tYSmr2t5RXSbOrRRoTfTWP24xymNiKIZeOwJK1o9bVDaJOwfiqdJzCRmcQpsgMKphRRSLMIrRK2MPllGd3hXnRNdsO0+ACPPkRx4O4h1seqQT7razATgsZTxmX6stYqRpu/A7FcHHxSdvWgAGfDPfCng7gx14ehpA321RHcStzBwtaLyoGPN2eZEbSX54gZtLzL0au4nehH//MB9M3fhG3FDtVIQ2lTKJA2mJ6A3PYm4rkniLyfwerXFOvl4XtVzJCj5hrwyVhtFjADKzop1D1DwIj2QCfaAh3MqJ3tRINuJYVDXflZpn2hBu3E3rLKxX+4vvb7gt3vX8AvdXpaCCj9EUYAAAAASUVORK5CYII="
          style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
        />
        <img
          alt="Built for scale"
          data-aos="fade-up"
          data-gatsby-image-ssr=""
          data-main-image=""
          data-src="/static/285b5b87d3c327c3ce8d379e9c86f131/a8b52/tech_1_52cd05fc87.png"
          data-srcset="/static/285b5b87d3c327c3ce8d379e9c86f131/53973/tech_1_52cd05fc87.png 30w,/static/285b5b87d3c327c3ce8d379e9c86f131/7ab40/tech_1_52cd05fc87.png 60w,/static/285b5b87d3c327c3ce8d379e9c86f131/a8b52/tech_1_52cd05fc87.png 120w"
          decoding="async"
          loading="lazy"
          sizes="(min-width: 120px) 120px, 100vw"
          style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
        />
        <noscript />
        <script
          type="module"
        >
          const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}
        </script>
      </div>
    </div>
    <p
      data-aos="fade-up"
    >
      Built for scale
    </p>
  </div>
</div>
`;
